docs/checkit.js

```diff
@@ -777,55 +786,118 @@ CheckIt.prototype.validateInput = function ($form, $inputFormGroup, inputId) {
 
 CheckIt.prototype.validateForm = function (startDateString, endDateString) {
     "use strict";
+    // First check that input was actually given for title, startDate
+    // and endDate
     var validateTitle = this.validateInput(this.$fullForm, this.$titleFormGroup, 'calendarTitle'),
         
         validateStartDate = this.validateInput(this.$fullForm, this.$startDateFormGroup, 'startDate'),
         
         validateEndDate = this.validateInput(this.$fullForm, this.$endDateFormGroup, 'endDate'),
     
         isValid = validateTitle && validateStartDate && validateEndDate;
-    
+
     // Make sure input is OK before parsing and validating dates
     return (isValid && this.validateDates(startDateString, endDateString, this.$endDateFormGroup));
    
 };
 
-CheckIt.prototype.createCalendar = function () {
-    // Create a new calendar and display it.
+CheckIt.prototype.createCalendar = function (calState) {
+    // creates a calendar
+    
     "use strict";
-    var title = this.$calendarTitle.val(),
-        start = this.$startDate.val(),
-        end = this.$endDate.val(),
-        state,
-        calendar,
+    var calendar,
         initP,
         buildP;
-        
+    //if (this.validateForm(start, end)) {
     
-    if (this.validateForm(start, end)) {
-        
-        //clear the previously displayed calendar
-        this.clearCalendarDiv();
+    //clear the previously displayed calendar
+    this.clearCalendarDiv();
 
-        //make a calendar State
-        state = emptyCalendarState({startDate: start, endDate: end, calendarTitle: title});
-        
-        //make calendar object
+    //make calendar object
 
-        calendar = new Calendar(state);
-        
-        // Initialize calendar in the storage
-        initP = this.store.initializeCalendar(calendar);
+    calendar = new Calendar(calState);
+
+    // Initialize calendar in the storage
+    initP = this.store.initializeCalendar(calendar);
+
+    //add calendar to dropdown
+    this.addCalendarToDropdown(calendar.state.uniqueId, calendar.state.title,
+        this.$calendarDropdown);
 
-        //add calendar to dropdown
-        this.addCalendarToDropdown(calendar.state.uniqueId, calendar.state.title,
-            this.$calendarDropdown);
+    //build the calendar
+    buildP = this.buildCalendar(calendar);
+    this.hideForm(this.$buildCalendarForm);
     
-        //build the calendar
-        buildP = this.buildCalendar(calendar);
-        this.hideForm(this.$buildCalendarForm);
+    return Promise.all([initP, buildP]);
+    //}
+};
+
+CheckIt.prototype.editCalendar = function (title, start, end, activeCalId) {
+    // Edits the calendar currently on display (current active calendar)
+    // and displays the edited version
+    "use strict";
+    
+    var checkit = this,
+        startDate = moment(start, "YYYY-MM-DD"),
+        endDate = moment(end, "YYYY-MM-DD"),
+        loadP;
+    
+    // I want to return a non-pending promise
```

Comment:
What do you mean here? Promises are pending until they resolve. You don't decide whether or not it is pending. It will be pending until the work is done.

Reply:
I don't know why I wrote that.

Reply:
Why return a promise there at all?

Reply:
Because the caller is probably expecting this function to return a Promise, I would imagine? If so, then if you don't return a Promise it will break when it tries to resolve it.